app/controllers/api/instances_controller.rb

```diff
@@ -140,8 +140,12 @@ def pause_instance(instance)
     end
 
     def validate_instance_for_action(instance, action)
-      validation = instance.send("validate_#{action}")
-      action_result(validation[:available], validation[:message].to_s)
+      if instance.respond_to?("supports_#{action}?")
+        action_result(instance.public_send("supports_#{action}?"), instance.unsupported_reason(action.to_sym))
```

Comment:
@Fryguy if not, I would personally lean towards extracting a service object to handle this stuff....I've been thinking of extracting some services to help lower the total surface area of `base_controller.rb` &co.